sallyport/src/lib.rs

```diff
@@ -179,24 +179,26 @@ impl Block {
     }
 
     /// Returns a Cursor for the Block
-    pub fn cursor(&mut self) -> Cursor<'_> {
-        Cursor(UnsafeCell::new(&mut self.buf))
+    pub fn cursor(&mut self) -> Cursor {
+        Cursor(&mut self.buf)
     }
 }
 
 /// Helper for allocation of untrusted memory in a Block.
-pub struct Cursor<'a>(UnsafeCell<&'a mut [u8]>);
+pub struct Cursor<'a>(&'a mut [u8]);
 
-impl Cursor<'_> {
+impl<'a> Cursor<'a> {
     /// Allocates an array, containing count number of T items. The result is uninitialized.
     ///
     /// # Safety
     ///
     /// This function is unsafe because it returns an uninitialized array.
-    pub unsafe fn alloc<T>(&self, count: usize) -> core::result::Result<&mut [T], ()> {
-        let inner_ptr = self.0.get();
+    pub unsafe fn alloc<T>(
+        self,
+        count: usize,
+    ) -> core::result::Result<(Cursor<'a>, &'a mut [T]), ()> {
```

Comment:
```suggestion
    ) -> core::result::Result<(Cursor<'_>, &mut [T]), ()> {
```

I think explicit elision should work here. If not, ignore this comment and the previous one.

Reply:
It is legal in Rust to return a lifetime-annotated struct with an anonymous lifetime: [play.rust-lang.org](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ae185c6940a47a2bc6681a9fc7863de3).